packages/camp/package.py

```diff
@@ -107,20 +107,28 @@ def blt_link_helpers(options, spec, spec_compiler):
             "/usr/tce/packages/gcc/gcc-4.9.3/lib64;/usr/tce/packages/gcc/gcc-4.9.3/gnu/lib64/gcc/powerpc64le-unknown-linux-gnu/4.9.3;/usr/tce/packages/gcc/gcc-4.9.3/gnu/lib64;/usr/tce/packages/gcc/gcc-4.9.3/lib64/gcc/x86_64-unknown-linux-gnu/4.9.3"))
 
     compilers_using_toolchain = ["pgc++", "xlc++", "xlC_r", "icpc", "clang++", "icpx"]
-    if any(compiler in spec_compiler.cxx for compiler in compilers_using_toolchain):
+    if any(tc_comp in compiler.cxx for tc_comp in compilers_using_toolchain):
         if spec_uses_toolchain(spec) or spec_uses_gccname(spec):
 
             # Ignore conflicting default gcc toolchain
             options.append(cmake_cache_string("BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE",
             "/usr/tce/packages/gcc/gcc-4.9.3/lib64;/usr/tce/packages/gcc/gcc-4.9.3/gnu/lib64/gcc/powerpc64le-unknown-linux-gnu/4.9.3;/usr/tce/packages/gcc/gcc-4.9.3/gnu/lib64;/usr/tce/packages/gcc/gcc-4.9.3/lib64/gcc/x86_64-unknown-linux-gnu/4.9.3"))
 
-    if "cce" in spec_compiler.cxx:
-        # Here is where to find libs that work for fortran
-        libdir = "/opt/cray/pe/cce/{0}/cce-clang/x86_64/lib".format(spec_compiler.version)
+    if "cce" in compiler.cxx:
         description = (
             "Adds a missing rpath for libraries " "associated with the fortran compiler"
         )
-        linker_flags = "${BLT_EXE_LINKER_FLAGS} -Wl,-rpath," + libdir
+        # Here is where to find libs that work for fortran
+        libdir = "/opt/cray/pe/cce/{0}/cce-clang/x86_64/lib".format(compiler.version)
+        linker_flags = "${{BLT_EXE_LINKER_FLAGS}} -Wl,-rpath,{0}".format(libdir)
+
+        version = "{0}".format(compiler.version)
+
+        if version == "16.0.0":
+            # Here is another directory added by cce@16.0.0
+            libdir = os.path.join(libdir,"x86_64-unknown-linux-gnu")
+            linker_flags += " -Wl,-rpath,{0}".format(libdir)
+
```

Comment:
@davidbeckingsale here is a change to workaround a specifics of cee@16.0.0: it has libunwind in an unusual subdirectory.
I asked LC, they say asked the vendor:
```
From the vendor:
this is likely a build (or maybe a packaging) issue. The x86_64-unknown-linux-gnu subdirectory does not appear prior to cce/16.0.0
```

No fix so far, so I workaround and add dedicated logic...